pygaggle/rerank/transformer.py

```diff
@@ -30,34 +31,78 @@
-                 tokenizer: QueryDocumentBatchTokenizer = None,
-                 use_amp = False):
-        self.model = model or self.get_model()
-        self.tokenizer = tokenizer or self.get_tokenizer()
+    def __init__(self, 
+                 pretrained_model_name_or_path: str  = 'castorini/monot5-base-msmarco',
+                 use_amp = False,
+                 token_false = None,
+                 token_true  = None):
+        self.model = self.get_model(pretrained_model_name_or_path)
+        self.tokenizer = self.get_tokenizer(pretrained_model_name_or_path)
+        self.token_false, self.token_true = token_false or self.get_prediction_tokens(
+                pretrained_model_name_or_path, self.tokenizer, token_false, token_true)
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
         self.device = next(self.model.parameters(), None).device
         self.use_amp = use_amp
+        self.token_false = token_false
+        self.token_true  = token_true
 
     @staticmethod
-    def get_model(pretrained_model_name_or_path: str = 'castorini/monot5-base-msmarco',
+    def get_model(pretrained_model_name_or_path: str,
                   *args, device: str = None, **kwargs) -> T5ForConditionalGeneration:
         device = device or ('cuda' if torch.cuda.is_available() else 'cpu')
         device = torch.device(device)
-        return T5ForConditionalGeneration.from_pretrained(pretrained_model_name_or_path,
+        return AutoModelForSeq2SeqLM.from_pretrained(pretrained_model_name_or_path,
                                                           *args, **kwargs).to(device).eval()
 
     @staticmethod
-    def get_tokenizer(pretrained_model_name_or_path: str = 't5-base',
+    def get_tokenizer(pretrained_model_name_or_path: str,
                       *args, batch_size: int = 8, **kwargs) -> T5BatchTokenizer:
         return T5BatchTokenizer(
             AutoTokenizer.from_pretrained(pretrained_model_name_or_path, use_fast=False, *args, **kwargs),
             batch_size=batch_size
         )
+    @staticmethod
+    def get_prediction_tokens(pretrained_model_name_or_path: str,
+            tokenizer, token_false, token_true):
+        if not token_false:
+            if pretrained_model_name_or_path in prediction_tokens:
+                token_false, token_true = prediction_tokens[pretrained_model_name_or_path]
+                token_false = tokenizer.tokenizer.get_vocab()['▁' + token_false]
+                token_true  = tokenizer.tokenizer.get_vocab()['▁' + token_true]
+                return (token_false, token_true)
+            else:
+                raise Exception("We don't know the indexes for the non-relevant/relevant tokens for\
+                        the checkpoint {pretrained_model_name_or_path} and you did not provide any.")
+
+
 
     def rescore(self, query: Query, texts: List[Text]) -> List[Text]:
+        '''
+        if not self.token_false:
```

Comment:
move this code to the __init__ since these tokens shouldn't change once the model is initialized?